src/test/java/com/softserve/sportshub/comment/CommentServiceImplTest.java

```diff
@@ -0,0 +1,145 @@
+
+class CommentServiceImplTest {
+
+    private final CommentDao commentDao = new InMemoryCommentDao();
+    UserDao userDao = Mockito.mock(UserDao.class);
+    private final CommentMapper mapper = new CommentMapper();
+
+    private final CommentServiceImpl commentService
+            = new CommentServiceImpl(commentDao, userDao, mapper);
+
+    private User user;
+
+    @BeforeEach
+    void setUp() {
+        user = new User(1L, "Lena", "password", false, false, new ArrayList<>());
+    }
+
+    @Test
+    public void shouldCreateCommentTest() {
+        // given
+        CreateCommentDto createCommentDto = new CreateCommentDto("Test comment");
+        Mockito.when(userDao.findByUsername(any())).thenReturn(user);
+        // when
+        ReadCommentDto createdComment = commentService.save(createCommentDto, "Lena");
+        // then
+        assertThat(createdComment.getContent()).isEqualTo("Test comment");
+        assertThat(createdComment.getCreatedAt()).isEqualTo(LocalDate.now().toString());
+        assertThat(createdComment.getOwner()).isEqualTo(user.getUsername());
+    }
+
+    @Test
+    public void shouldReturnAllComments() {
+        // given
+        testInMemoryDaoInitialized();
+
+        Comment comment1 = createNewComment(1L, "Test comment");
+        Comment comment2 = createNewComment(2L, "Second test comment");
+        List<ReadCommentDto> commentsExpected
+                = Stream.of(comment1, comment2)
+                        .map(mapper::entityToReadDto)
+                        .toList();
+
+        // when
+        List<ReadCommentDto> commentsFound = commentService.list();
+
+        //then
+        assertThat(commentsFound).hasSameSizeAs(commentsExpected);
+        assertThat(commentsFound).element(0).usingRecursiveComparison().isEqualTo(commentsExpected.get(0));
+        assertThat(commentsFound)
+                .element(1)
+                .hasFieldOrPropertyWithValue("content", commentsExpected.get(1).getContent());
+    }
+
+    @Test
+    public void givenCorrectId_whenFindByIdIsCalled_shouldFindOneComment() {
+        // given
+        testInMemoryDaoInitialized();
+        Comment expectedComment = createNewComment(2L, "Second test comment");
+        // when
+        Comment actualComment = commentService.findById(2L);
+        // then
+        assertThat(actualComment).isNotNull();
+        assertThat(actualComment).usingRecursiveComparison().isEqualTo(expectedComment);
+
+    }
+
+    @Test
+    public void givenCorrectId_whenDeleteIsCalled_shouldDeleteTheComment() {
+        // given
+        List<Comment> comments = testInMemoryDaoInitialized();
+        // when
+        // then
+        assertThat(commentService.delete(500L)).isFalse();
+        assertThat(commentService.delete(1L)).isTrue();
+        assertThat(commentService.list()).hasSize(1);
+        assertThat(commentService.findById(2L)).hasFieldOrPropertyWithValue("content","Second test comment");
```

Comment:
I'd rather use the getter to access the value:
`assertThat(commentService.findById(2L).getContent()).isEqualTo("Second test comment");`
In this case the compiler won't help you if you misstype "content" or someone will have to change the field name in the future